src/notification/repository/webhook-repository.service.ts

```diff
@@ -0,0 +1,71 @@
+import { NotFoundException } from '@nestjs/common';
+import { catchError, map, Observable } from 'rxjs';
+import { Reference } from '../../domain/reference';
+import { CouchDbClient } from '../../couchdb/couch-db-client.service';
+import { CouchDbRow, CouchDbRows, DocSuccess } from '../../couchdb/dtos';
+
+export interface WebhookEntity {
+  id: string;
+  label: string;
+  target: {
+    method: 'GET' | 'POST';
+    url: string;
+  };
+  authentication: {
+    type: 'API_KEY';
+    apiKey: {
+      iv: string;
+      data: string;
+    };
+  };
+  owner: {
+    type: 'USER'; // TODO: group support?
+    id: string;
+  };
+  reportSubscriptions: Reference[];
+}
+
+export class WebhookRepository {
+  constructor(private http: CouchDbClient) {}
+
+  fetchAllWebhooks(): Observable<WebhookEntity[]> {
+    return this.http
```

Comment:
```suggestion
    // TODO: change this to use a configured CouchDbClient here, to simplify and remove duplication
    return this.http
```

Or is there a reason to do this separate?